R/landing_popup_module.R

```diff
@@ -0,0 +1,118 @@
+#' @param content The content of the popup. Passed to `...` of `shiny::modalDialog`. Can be a `character`
+#' or a text input control (like `textInput`) or a list of `shiny` tags. See examples.
+#' @param buttons `shiny.tag` or a list of tags (`tagList`). Typically a `modalButton` or `actionButton`. See examples.
+#'
+#' @return A `teal_module` (extended with `teal_landing_module` class) to be used in `teal` applications.
+#'
+#' @examples
+#' app1 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       content = "A place for the welcome message or a disclaimer statement.",
+#'       buttons = modalButton("Proceed")
+#'     ),
+#'     module(
+#'       label = "example module",
+#'       server = function(input, output, session, data) {},
+#'       ui = function(id, ...) div(p("Example text"))
+#'     )
+#'   )
+#' )
+#' if (interactive()) {
+#'   shinyApp(app1$ui, app1$server)
+#' }
+#'
+#' app2 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       title = "Welcome",
+#'       content = tags$b(
+#'         "A place for the welcome message or a disclaimer statement.",
+#'         style = "color: red;"
+#'       ),
+#'       buttons = tagList(
+#'         modalButton("Proceed"),
+#'         actionButton("close", "Read more",
+#'           onclick = "window.open('http://google.com', '_blank')"
+#'         )
+#'       )
+#'     ),
+#'     module(
+#'       label = "example module",
+#'       server = function(input, output, session, data) {},
+#'       ui = function(id, ...) div(p("Example text"))
+#'     )
+#'   )
+#' )
+#'
+#' if (interactive()) {
+#'   shinyApp(app2$ui, app2$server)
+#' }
+#'
+#' app3 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       title = "Welcome",
+#'       content = "App will close, once you click the rejection button!",
+#'       buttons = tagList(
+#'         modalButton("Proceed"),
+#'         actionButton("close", "Reject", onclick = "window.close()")
+#'       )
+#'     ),
+#'     module(
+#'       label = "example module",
+#'       server = function(input, output, session, data) {},
+#'       ui = function(id, ...) div(p("Example text"))
+#'     )
+#'   )
+#' )
+#'
+#' if (interactive()) {
+#'   shinyApp(app3$ui, app3$server, options = list(launch.browser = TRUE))
+#' }
+#'
+#' @export
+landing_popup_module <-
+  function(label = "Landing Popup",
+           title = NULL,
+           content = NULL,
+           buttons = modalButton("Accept")) {
+    checkmate::assert_string(label)
+    checkmate::assert_string(title, null.ok = TRUE)
+    checkmate::assert_multi_class(
+      content,
+      classes = c("character", "shiny.tag", "shiny.tag.list", "html"), null.ok = TRUE
+    )
+    checkmate::assert_multi_class(buttons, classes = c("shiny.tag", "shiny.tag.list"))
```

Comment:
Consider adding assertion on elements of buttons having `<element>$attribs$type == "button"`.